### PR TITLE
Port file upload component to TS.

### PR DESCRIPTION
**Pull request by pavel-mironchik**

### CODE REVIEW

File: src/Components/Misc/FileUpload.vue [hunk 1]
```diff
@@ -54,145 +54,124 @@
...
+
+            this.errorMessage = response.errors[0].title;
+        }
+
+        /**
+         * Handle the success event.
+         */
+        handleSuccess(file: any, response: any)
+        {
+            this.$store.commit('files/ADD', response);
```
Comment (pavel-mironchik) [inline src/Components/Misc/FileUpload.vue]: I'm not sure about this particular line. Do we need to import Files module from our application here?

Comment (pavel-mironchik) [reply src/Components/Misc/FileUpload.vue]: I mean add something like 
```
import { File } from '../../../../../resources/ts/App/State/Models';
```
and then do `File.insert(response);`

Comment (silvioiannone) [reply src/Components/Misc/FileUpload.vue]: Yes, we should strive to use the VuexORM models whenever possible.

**Comment by pavel-mironchik** [comment]
I tested an approach with VuexORM model in the Boptimera application, it works as expected - uploaded file was added into Documents datatable view (which uses VuexORM to query documents array).

**Pull request merged**